base/main.cpp

```diff
@@ -547,14 +548,17 @@ extern "C" int scummvm_main(int argc, const char * const argv[]) {
 
 		// Try to find a MetaEnginePlugin which feels responsible for the specified game.
 		const Plugin *plugin = detectPlugin();
-		if (plugin) {
-			// Unload all plugins not needed for this game, to save memory
 
+		// Then, get the relevant Engine plugin from MetaEngine.
+		const Plugin *enginePlugin = nullptr;
+		if (plugin)
+			enginePlugin = PluginMan.getEngineFromMetaEngine(plugin);
+
+		if (plugin && enginePlugin) {
```

Comment:
You just checked for null values for plugin above. Just checking for enginePlugin here should suffice

Reply:
OK, done.